### PR TITLE
Replace DNSCrypt with DNSCrypt-Proxy

### PR DESCRIPTION
**Pull request by nitrohorse**
## Description

Resolves: https://github.com/privacytoolsIO/privacytools.io/issues/684

#### Check List 

- [x] I have read and understand [the contributing guidelines](https://github.com/privacytoolsIO/privacytools.io/blob/master/.github/CONTRIBUTING.md).

- [x] I have [listed the source code](https://github.com/privacytoolsIO/privacytools.io/blob/master/source_code.md).

- [x] This project is [free/libre software](https://www.wikipedia.org/wiki/Free_software).

- [x] This project has an [associated discussion](https://github.com/privacytoolsIO/privacytools.io/issues).

* Netlify preview for the mainly edited page: https://deploy-preview-1254--privacytools-io.netlify.com/providers/dns/#dns

* Code Repository (if applicable): https://github.com/jedisct1/dnscrypt-proxy

### CODE REVIEW

**Comment by nitrohorse** [comment]
> Is it just me or doesn't the dnscrypt proxy not have an icon yet?

Lol wow just looking in their repo: https://github.com/jedisct1/dnscrypt-proxy/blob/master/logo.svg

File: _includes/sections/dns.html [hunk 1]
```diff
@@ -1,30 +1,37 @@
...
-image="/assets/img/tools/DNSCrypt.png"
-description="A protocol for securing communications between a client and a DNS resolver. The DNSCrypt protocol uses high-speed high-security elliptic-curve cryptography and is very similar to DNSCurve, but focuses on securing communications between a client and its first-level resolver."
-website="https://dnscrypt.info/"
-forum="https://forum.privacytools.io/t/discussion-dnscrypt/340"
-github="https://github.com/jedisct1/dnscrypt-proxy"
+{%
+  include cardv2.html
+  title="DNSCrypt-Proxy - Local DNS Server"
+  image="/assets/img/tools/DNSCrypt-Proxy.png"
+  description="DNSCrypt-Proxy is a command-line proxy that bridges applications expecting regular DNS with servers supporting encrypted DNS (DNSCrypt and DNS over HTTPS). Can also display the DNS activity, cache results to improve speed, and locally block unwanted content."
```
Comment (Mikaela) [inline _includes/sections/dns.html]: I am not sure on this description.

* I cannot remember giving DNSCrypt-proxy any direct CLI commands.
* I think "bridges applications expecting regular DNS with servers supporting encrypted DNS" suggests that every app should manage DNS by itself, probably even DNS over HTTPS and centralize queries somewhere out of user control.
* "Can also display the DNS activity" - you mean can log DNS queries or am I unaware of some CLI?
* "locally block unwanted content." - I guess so,  but I think there would be a better word that would also include forcing safe search etc. I don't know what the word would be though.

File: _includes/sections/dns.html [hunk 2]
```diff
@@ -1,30 +1,37 @@
...
+  website="https://github.com/jedisct1/dnscrypt-proxy/"
+  forum="https://forum.privacytools.io/t/discussion-dnscrypt-proxy/1498"
+  github="https://github.com/jedisct1/dnscrypt-proxy/"
+  linux="https://github.com/jedisct1/dnscrypt-proxy/wiki/installation"
+  mac="https://github.com/jedisct1/dnscrypt-proxy/wiki/installation"
+  android="https://github.com/jedisct1/dnscrypt-proxy/wiki/installation"
+  ios="https://github.com/jedisct1/dnscrypt-proxy/wiki/installation"
```
Comment (Mikaela) [inline _includes/sections/dns.html]: The `operatingsystem="link"` things won't actually do anything as you can see in the preview, so I think they should be removed.

File: _includes/sections/dns.html [hunk 3]
```diff
@@ -1,30 +1,37 @@
...
-{% include cardv2.html
-title="DNSCrypt - Tool"
-image="/assets/img/tools/DNSCrypt.png"
-description="A protocol for securing communications between a client and a DNS resolver. The DNSCrypt protocol uses high-speed high-security elliptic-curve cryptography and is very similar to DNSCurve, but focuses on securing communications between a client and its first-level resolver."
-website="https://dnscrypt.info/"
-forum="https://forum.privacytools.io/t/discussion-dnscrypt/340"
-github="https://github.com/jedisct1/dnscrypt-proxy"
+{%
+  include cardv2.html
+  title="DNSCrypt-Proxy - Local DNS Server"
```
[the remark this answers is not in the document]
Comment (nitrohorse) [reply _includes/sections/dns.html]: Same, should we leave off a tag line for this?

File: _includes/sections/dns.html (hunk 2 above)
Comment (nitrohorse) [reply _includes/sections/dns.html]: Gotcha, that makes sense. Wasn’t sure if we want these filled out even in these cases but will remove them.

File: _includes/sections/dns.html (hunk 1 above)
Comment (nitrohorse) [reply _includes/sections/dns.html]: > - I cannot remember giving DNSCrypt-proxy any direct CLI commands.

It's possible, yeah:

dnscrypt-proxy1

> - I think "bridges applications expecting regular DNS with servers supporting encrypted DNS" suggests that every app should manage DNS by itself, probably even DNS over HTTPS and centralize queries somewhere out of user control.
> - "Can also display the DNS activity" - you mean can log DNS queries or am I unaware of some CLI?
> - "locally block unwanted content." - I guess so, but I think there would be a better word that would also include forcing safe search etc. I don't know what the word would be though.

Hmm what about this description (taken in part from [Arch's wiki](https://wiki.archlinux.org/index.php/Dnscrypt-proxy
) and updated):
> DNSCrypt-Proxy is a command-line DNS proxy with support for the encrypted DNS protocols, DNS over HTTPS and DNSCrypt. Can cache results to improve speed, allows filtering, forwarding, and cloaking, and is compatible with DNSSEC.

Comment (Mikaela) [reply _includes/sections/dns.html]: I like it, other than the last part

> and is compatible with DNSSEC.

I don't know what they mean by "compatible with DNSSEC" unless it means "you can configure dnscrypt-proxy to require DNSSEC capable servers from sources file" as there is the "but" that dnscrypt-proxy won't validate DNSSEC by itself.

Comment (nitrohorse) [reply _includes/sections/dns.html]: Yeah, I just see that's a bit odd wording...

Comment (nitrohorse) [reply _includes/sections/dns.html]: I can just leave it out.

**Pull request merged**